src/menu/src/Menu.js

```diff
@@ -8,89 +8,89 @@ import MenuOption from './MenuOption'
 
-          if (goingDown) {
-            if (indexOfItem < menuItems.current.length - 1) {
-              return menuItems.current[indexOfItem + 1]
-            }
+    const focusNext = (currentItem, startItem) => {
 
-            return startItem
-          }
+      // Determine which item is the startItem (first or last)
+      const goingDown = startItem === firstItem.current
+
+      // Helper function for getting next legitimate element
+      const move = elem => {
+        const indexOfItem = menuItems.current.indexOf(elem)
 
-          if (indexOfItem - 1 > -1) {
-            return menuItems.current[indexOfItem - 1]
+        if (goingDown) {
+          if (indexOfItem < menuItems.current.length - 1) {
+            return menuItems.current[indexOfItem + 1]
           }
 
           return startItem
         }
 
-        // Make first move
-        let nextItem = move(currentItem)
-
-        // If the menuitem is disabled move on
-        while (nextItem.disabled) {
-          nextItem = move(nextItem)
+        if (indexOfItem - 1 > -1) {
+          return menuItems.current[indexOfItem - 1]
         }
 
-        // Focus the first one that's not disabled
-        nextItem.focus()
+        return startItem
       }
 
-      menuItems.current.forEach(menuItem => {
-        // Handle key presses for menuItem
-        menuItem.addEventListener('keydown', e => {
-          // Go to next/previous item if it exists
-          // or loop around
+      // Make first move
+      const nextItem = move(currentItem)
 
-          if (e.key === 'ArrowDown') {
-            e.preventDefault()
-            focusNext(menuItem, firstItem.current)
-          }
+      // Focus the first one that's not disabled
+      nextItem.focus()
+    }
 
-          if (e.key === 'ArrowUp') {
-            e.preventDefault()
-            focusNext(menuItem, lastItem.current)
-          }
+    function onKeyPressListener(e) {
+      // Go to next/previous item if it exists
+      // or loop around
 
-          if (e.key === 'Home') {
-            e.preventDefault()
-            firstItem.current.focus()
-          }
+      const menuItem = e.target
+      if (e.key === 'ArrowDown') {
+        e.preventDefault()
+        focusNext(menuItem, firstItem.current)
+      }
 
-          if (e.key === 'End') {
-            e.preventDefault()
-            lastItem.current.focus()
-          }
-        })
-      })
+      if (e.key === 'ArrowUp') {
+        e.preventDefault()
+        focusNext(menuItem, lastItem.current)
+      }
+
+      if (e.key === 'Home') {
+        e.preventDefault()
+        firstItem.current.focus()
+      }
+
+      if (e.key === 'End') {
+        e.preventDefault()
+        lastItem.current.focus()
+      }
+    }
+
+    menuItems.current.forEach(menuItem => menuItem.addEventListener('keydown', onKeyPressListener))
+
+    return () => {
+      menuItems.current.forEach(menuItem => menuItem.removeEventListener('keydown', onKeyPressListener))
```

Comment:
Not in love with adding event listeners to every element just to capture keypresses. Seems like we could have a single event listener added and remove when the menu is opened/closed?

(realize this is out of scope for your changes, but just seeing this with fresh eyes)

Reply:
I agree, we could use [event delegation](https://davidwalsh.name/event-delegate) here to only register a single event listener on the parent. will look into this.

Reply:
No worries about solving here in this PR. We can do this in a follow-up (if you are interested in taking a stab at it!)